### PR TITLE
feat: dataset example slideover

### LINKED ISSUE #3302: [datasets] [ui] view example slideover
Screenshot 2024-05-23 at 4 21 27 PM

### PR DESCRIPTION
**Pull request by axiomofjoy**
Screenshot 2024-05-28 at 10 15 37 PM

resolves #3302

### CODE REVIEW

File: app/src/pages/trace/TracePage.tsx [hunk 1]
```diff
@@ -503,7 +503,7 @@ function AddSpanToDatasetButton({ span }: { span: Span }) {
         }}
       />
     );
-  }, [span]);
+  }, [span, notifySuccess]);
```
Comment (axiomofjoy) [inline app/src/pages/trace/TracePage.tsx:506]: Fixes lint error

**Review by RogerHYang — APPROVED** [review]

File: app/src/pages/dataset/DatasetExamplePage.tsx [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+  backgroundColor: "light",
+  borderColor: "light",
+  variant: "compact",
+  collapsible: true,
+};
+
+/**
+ * A block of JSON content that is not editable.
+ */
+export function JSONBlock({ children }: { children: string }) {
```
Comment (mikeldking) [inline app/src/pages/dataset/DatasetExamplePage.tsx:108]: I think ultimately I'm probably ready to converge on a single component for all this at this point (https://github.com/Arize-ai/phoenix/blob/datasets/app/src/components/code/JSONEditor.tsx)

But I can do this refactor in a follow-up

File: app/src/pages/dataset/DatasetExamplesTable.tsx [hunk 3]
```diff
@@ -142,7 +144,12 @@ export function DatasetExamplesTable({
         ) : (
           <tbody>
             {rows.map((row) => (
-              <tr key={row.id}>
+              <tr
+                key={row.id}
+                onClick={() => {
+                  navigate(`examples/${row.original.id}`);
```
[the remark this answers is not in the document]
Comment (axiomofjoy) [reply app/src/pages/dataset/DatasetExamplesTable.tsx:154]: addressed in e8e2f5691f96920ccf47da1446bc0071c740db59

File: app/src/pages/dataset/DatasetExamplePage.tsx [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+      </Dialog>
+    </DialogContainer>
+  );
+}
+
+const defaultCardProps: Partial<CardProps> = {
+  backgroundColor: "light",
+  borderColor: "light",
+  variant: "compact",
+  collapsible: true,
```
[the remark this answers is not in the document]
Comment (axiomofjoy) [reply app/src/pages/dataset/DatasetExamplePage.tsx:102]: addressed in [88c7d5c](https://github.com/Arize-ai/phoenix/pull/3325/commits/88c7d5cbecac20ca43f6cb3544a38f527230de39)

**Review by mikeldking — APPROVED** [review]

**Pull request merged**